app/src/main/java/com/framgia/quangtran/music_42/ui/homescreen/adapters/GenresAdapter.java

```diff
@@ -0,0 +1,53 @@
+package com.framgia.quangtran.music_42.ui.homescreen.adapters;
+
+import android.support.annotation.NonNull;
+import android.support.v7.widget.RecyclerView;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import com.framgia.quangtran.music_42.R;
+import com.framgia.quangtran.music_42.data.model.Genre;
+
+import java.util.List;
+
+public class GenresAdapter extends RecyclerView.Adapter<GenresAdapter.ViewHolder> {
+    private List<Genre> mGenres;
+
+    public GenresAdapter(List<Genre> genres) {
+        this.mGenres = genres;
+    }
+
+    @NonNull
+    @Override
+    public GenresAdapter.ViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
+        LayoutInflater inflater = LayoutInflater.from(viewGroup.getContext());
+        View contactView = inflater.inflate(R.layout.item_recycler_genres_home, viewGroup,
+                false);
+        return new GenresAdapter.ViewHolder(contactView);
+    }
+
+    @Override
+    public void onBindViewHolder(@NonNull GenresAdapter.ViewHolder viewHolder, int i) {
+        viewHolder.mImageGenre.setImageResource(mGenres.get(i).getImageUrl());
```

Comment:
34, 45 nên đưa xuống ViewHolder để xử lý để đảm bảo tính đóng gói.
```suggestion
        viewHolder.binData(mGenres.get(i));
```